### ISSUE TITLE
训练集与测试集划分的问题

### ISSUE THREAD

**Issue by wisest-cui** [zhouhaoyi/Informer2020#504]
作者您好！感谢您过去的工作，但是我对代码中的训练集、验证集、测试集有一些疑问。我不理解是否 您的测试集是选用了训练集的一部分？
在代码中，您在exp_informer.py中 约 129行 处的train函数进行训练任务。
def train(self, setting):
        train_data, train_loader = self._get_data(flag = 'train')
        vali_data, vali_loader = self._get_data(flag = 'val')
        test_data, test_loader = self._get_data(flag = 'test')

在这里加载训练集、验证集、测试集分别使用了三次self._get_data，仅用字段flag进行区分。
在__get_data函数中：
    data_set = Data(
          root_path=**args.root_path,**
          data_path=**args.data_path,**
          flag=flag,
          size=[args.seq_len, args.label_len, args.pred_len],
          features=args.features,
          target=args.target,
          inverse=args.inverse,
          timeenc=timeenc,
          freq=freq,
          cols=args.cols
     
      print(flag, len(data_set))
      data_loader = DataLoader(
          data_set,
          batch_size=batch_size,
          shuffle=shuffle_flag,
          num_workers=args.num_workers,
          drop_last=drop_last)

        return data_set, data_loader

您同样传入 **同一个csv文件**，而且除了shuffle变量标记是否打乱以外，好像并没有区分训练集和测试集。
因此我不理解是否 您的测试集是选用了训练集的一部分？
如果并非这样，希望您可以给予指导，指出训练集与测试集的划分部分是通过什么实现的？
祝科研顺利、身体健康！

**Comment by wisest-cui** [comment]
此问题已解决，通过 __get_data函数中的 dataset中的flag变量拆分了训练集、测试集。
代码中的 训练集固定选用最开始360天的数据 测试集选用360-480天的数据 验证集选用480-600天的数据。
这一点其他人复现论文并且接入自己的数据集时，需要注意 改成自己数据集的长度！
